scripts/generate-vectors.ts

```diff
@@ -0,0 +1,188 @@
+import { bytesToHex } from '@metamask/utils';
+// eslint-disable-next-line import/no-nodejs-modules
+import { randomInt, randomBytes } from 'crypto';
+
+import {
+  BIP32Node,
+  BIP_32_HARDENED_OFFSET,
+  ed25519,
+  MAX_BIP_32_INDEX,
+  secp256k1,
+  SLIP10Node,
+} from '../src';
+import { Curve } from '../src/curves';
+import { createBip39KeyFromSeed } from '../src/derivers/bip39';
+
+/**
+ * Get a random boolean value.
+ *
+ * @returns A random boolean value.
+ */
+function getRandomBoolean() {
+  return randomInt(2) === 0;
+}
+
+/**
+ * Get a random BIP-32 index.
+ *
+ * @param hardened - Whether the index should be hardened. Defaults to a random
+ * boolean value.
+ * @returns A random BIP-32 index.
+ */
+function getRandomIndex(hardened: boolean = getRandomBoolean()) {
+  if (hardened) {
+    return randomInt(BIP_32_HARDENED_OFFSET, MAX_BIP_32_INDEX);
+  }
+
+  return randomInt(0, BIP_32_HARDENED_OFFSET);
+}
+
+/**
+ * Get a random seed.
+ *
+ * @param length - The length of the seed. Defaults to a random integer between
+ * 16 and 64 bytes.
+ * @returns A random seed.
+ */
+function getRandomSeed(length = randomInt(16, 64)) {
+  return randomBytes(length);
+}
+
+/**
+ * Get a random BIP-32 path, with the given length.
+ *
+ * @param spec - The path specification.
+ * @param length - The length of the path. Defaults to a random integer between
+ * 1 and 20.
+ * @param hardened - Whether the path should be hardened. Defaults to a random
+ * boolean value.
+ * @returns A random BIP-32 path.
+ */
+function getRandomPath(
+  spec: 'bip32' | 'slip10',
+  length = randomInt(1, 20),
+  hardened?: boolean,
+) {
+  return new Array(length).fill(0).map(() => {
+    const index = getRandomIndex(hardened);
+    const formattedIndex =
+      index >= BIP_32_HARDENED_OFFSET
+        ? `${index - BIP_32_HARDENED_OFFSET}'`
+        : index;
+
+    return `${spec}:${formattedIndex}` as BIP32Node;
+  });
+}
+
+/**
+ * Get a random key vector.
+ *
+ * @param node - The node to derive the keys from.
+ * @param spec - The path specification.
+ * @param hardened - Whether the keys should be hardened. Defaults to a random
+ * boolean value.
+ * @returns A random key vector.
+ */
+async function getRandomKeyVector(
+  node: SLIP10Node,
+  spec: 'bip32' | 'slip10',
+  hardened?: boolean,
+) {
+  const path = getRandomPath(spec, undefined, hardened);
+  const child = await node.derive(path);
+
+  return {
+    path: {
+      tuple: path,
+      string: path.join(' / '),
+    },
+    privateKey: child.privateKey,
+    publicKey: child.compressedPublicKey,
```

Comment:
Is there a reason we don't use the full length public key?

Reply:
Not particularly. The compressed public key is smaller so there's less data in the vectors file, but either way it's not a big difference. From my understanding, compressed public keys are essentially just the `x`-coordinate on the curve, so if you want the point on the curve you have to calculate the `y`-coordinate first. Uncompressed public keys are a combination of `(x, y)`, so slightly faster to work with, as you don't need to calculate anything.